healthchecks.go

```diff
@@ -0,0 +1,151 @@
+package cloudflare
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// Healthcheck describes a Healthcheck object.
+type Healthcheck struct {
+	ID            string                  `json:id,omitempty"`
+	CreatedOn     *time.Time              `json:"created_on,omitempty"`
+	ModifiedOn    *time.Time              `json:"modified_on,omitempty"`
+	Name          string                  `json:"name"`
+	Description   string                  `json:"description"`
+	Suspended     bool                    `json:"suspended"`
+	Address       string                  `json:"address"`
+	Retries       int                     `json:"retries,omitempty"`
+	Timeout       int                     `json:"timeout,omitempty"`
+	Interval      int                     `json:"interval,omitempty"`
+	Type          string                  `json:"type,omitempty"`
+	CheckRegions  []string                `json:"check_regions"`
+	HTTPConfig    *HealthcheckHTTPConfig  `json:"http_config,omitempty"`
+	TCPConfig     *HealthcheckTCPConfig   `json:"tcp_config,omitempty"`
```

Comment:
Wasn't 100% sure if this is the right approach. Wanted to ensure that when unspecified `TCPConfig` (or `HTTPConfig` is omitted entirely, rather than submitted to the API as an empty object, as the response from Cloudflare is different between the two:

Omitted:
```
"tcp_config": null,
```
Sent as empty block `{}`:
```
"tcp_config": {
  "method": "",
  "port": 0
},
```

TCP Healthchecks don't appear to be implemented yet as a feature however there are traces of them in the Healthcheck API objects

Reply:
For now, I'd probably leave the undocumented stuff out. We can raise a ticket and get clarification whether it's intended to be there or not.